### ISSUE TITLE
Allow user to format a specific substring.

### ISSUE THREAD

**Issue by AdamMc331** [androidessence/RichTextView#4]
Rather than limited to formatting spans by index, allow the user to format a certain substring.

**Comment by Mauker1** [comment]
You could use regex for that.

**Comment by AdamMc331** [comment]
Yeah. Since the SpannableString spans text based on Index, I'd have to write a method to determine the start and end index of the substring, for each occurence it appears.
